### ISSUE TITLE
Problem: OneDrive authorization does not work with selfhosted KeeWeb app.

### ISSUE THREAD

**Issue by victorvogelpoel** [keeweb/keeweb#358]
Problem: OneDrive authorization does not work with selfhosted KeeWeb app.

Actions: 
attempt to open keepass database from OneDrive. When clicking OneDrive button, a popup windows appears. It says almost immediately:
"We're unable to complete your request"
"Microsoft account is experiencing technical problems. Please try again later."
The URL explains the issue:
https://login.live.com/err.srf?lc=1033#error=invalid_request&error_description=The+provided+value+for+the+input+parameter+%27redirect_uri%27+is+not+valid.+The+expected+value+is+%27https://login.live.com/oauth20_desktop.srf%27+or+a+URL+which+matches+the+redirect+URI+registered+for+this+client+application.

Appversion
KeeWeb v1.3.1 (234333d, 2016-09-02)
Environment: web
User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0

Must I set a configuration JSON file with following entry?
""onedriveClientId": "https://login.live.com/oauth20_desktop.srf"

**Comment by antelle** [comment]
Yes, you have to create OneDrive app and, configure it and set its key in `onedriveClientId`

**Comment by antelle** [comment]
We should display a special error, like we do in Dropbox case...

**Comment by jaddie** [comment]
How should this be configured if it is for a onedrive for business account? I am still not sure how to set the redirect URI, as it seems to complain that it doesn't match or similar?

**Comment by antelle** [comment]
I don't think it works with onedrive business account.

**Comment by kcgaisford** [comment]
End of 2018 - Any update on KeeWeb usage with Office 365 / Onedrive for business?

**Comment by jmbarneck** [comment]
It would be great if this would work with Onedrive business accounts. What is the status?

**Comment by antelle** [comment]
Why are these questions about OneDrive for business asked in this issue? They're not related to the original report.
Enterprise OneDrive for business should be supported now, see https://github.com/keeweb/keeweb/issues/683. Please create a separate ticket if it doesn't work.
